front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -8,12 +8,71 @@ import type { NetzgrafikDto, NGEEvent, TrainrunSection, Node, Trainrun } from '.
 
 const createdTrainrun = new Map<number, number>();
 
-const getTrainrunSectionsByTrainrunId = (trainrunSections: TrainrunSection[], trainrunId: number) =>
-  trainrunSections.filter((section) => section.trainrunId === trainrunId);
-
 const getNodeById = (nodes: Node[], nodeId: number | string) =>
   nodes.find((node) => node.id === nodeId);
 
+const getTrainrunSectionsByTrainrunId = (netzgrafikDto: NetzgrafikDto, trainrunId: number) => {
+  // The sections we obtain here may be out-of-order. For instance, for a path
+  // A → B → C, we may get two sections B → C and then A → B. We need to
+  // re-order the section A → B before B → C.
+  const sections = netzgrafikDto.trainrunSections.filter(
+    (section) => section.trainrunId === trainrunId
+  );
+
+  // Sections are linked together with transitions and ports. A section in the
+  // middle of a path will have a dedicated target port connected to a
+  // transition on one side, and on the other side connected to a source port
+  // dedicated to the next section. Build a map of sections keyed by their
```

Comment:
I'm not sure, but I think the description would be clearer like this 
```suggestion
  // transition on one side, and on the other side connected to a source port
  // dedicated to the transition in the next node. Build a map of sections keyed by their
```

Reply:
Hm, I don't think that's quite correct: I'm only looking at a single node here and trying to explain that it goes like section → port → transition → port → section.

I've tried to make the paragraph clearer by including a small diagram - let me know if it's better or if you have ideas to improve it!